packages/frontend-core/src/components/grid/stores/datasources/viewV2.ts

```diff
@@ -1,16 +1,42 @@
 import { get } from "svelte/store"
-import { SortOrder } from "@budibase/types"
+import {
+  Row,
+  SaveRowRequest,
+  SortOrder,
+  UpdateViewRequest,
+} from "@budibase/types"
+import { Store as StoreContext } from ".."
 
 const SuppressErrors = true
 
-export const createActions = context => {
+interface ViewActions {
+  viewV2: {
+    actions: {
+      saveDefinition: (newDefinition: UpdateViewRequest) => Promise<void>
+      addRow: (row: SaveRowRequest) => Promise<Row>
+      updateRow: (row: SaveRowRequest) => Promise<Row>
+      deleteRows: (rows: (string | Row)[]) => Promise<void>
+      getRow: (id: string) => Promise<Row>
+      isDatasourceValid: (datasource: {
```

Comment:
Can we make a type for this - it looks quite different to a backend datasource given the `id` rather than `_id` and the `tableId` - would be good to document this in `@budibase/types` under `types/src/ui/stores` as a UIDatasource or something like that